include/grpc++/impl/codegen/status.h

```diff
@@ -48,10 +48,13 @@ class Status {
   Status() : code_(StatusCode::OK) {}
 
   /// Construct an instance with associated \a code and \a error_message
```

Comment:
Done.

An assert at this point may break existing users if they happen to do the wrong thing, but we might choose to do it at a later stage (after explicitly warning for a while).